http-protocol-binding.md

```diff
@@ -429,6 +429,48 @@ Content-Length: nnnn
 
 ```
 
+### 3.4. Json Streaming Content Mode
+
+In the _json streaming_ content mode several events are sent into a single HTTP
+request or response body using [RFC7464](https://www.rfc-editor.org/rfc/rfc7464.html).
+
+#### 3.4.1. HTTP Content-Type
+
+The [HTTP `Content-Type`][content-type] header MUST be set to the media type `application/cloudevents-stream`
+
+```text
+Content-Type: application/cloudevents-stream; charset=UTF-8
+```
+
+#### 3.4.2. Examples
+
+```text
+PUT /myresource HTTP/1.1
+Host: webhook.example.com
+Content-Type: application/cloudevents-stream; charset=UTF-8
```

Comment:
Maybe a newline delimited JSON (NDJSON) would be a better format here.

Elasticsearch could serve as an example of it used in a wild:
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-desc

Reply:
I've chosen the RFC7464 because it's a IETF spec, while the NDJSON is not a spec http://jsonlines.org/

Reply:
My concern is more basic... as someone who might write code to produce this stream, I'm much more likely to just do the equivalent of `printf("%s\n", eventJsonString)` and adding a RS is something I would either forget or be annoyed at having to figure out how to do.

I guess I just wondering, why it's not just any random whitespace since with at least one LF between the ending of one and the start of another - wouldn't that be the easiest for people to produce and parse?

Reply:
So there are a couple of things there to consider:

* As explained in the RFC7464: 

> In prose: any number of JSON texts, each encoded in UTF-8 [RFC3629],
   each preceded by one ASCII RS character, and each followed by a line
   feed (LF).  Since RS is an ASCII control character, it may only
   appear in JSON strings in escaped form (see [RFC7159]), and since RS
   may not appear in JSON texts in any other form, RS unambiguously
   delimits the start of any element in the sequence

The problem with newlines is that you could have it in the middle of a json (for example, if you're sending a pretty printed json), which makes the parsing a whole more complex. Using <RS><LF> is more simpler because you just scan your byte array and when you find those two consecutive chars you split the byte stream and parse the previous part as json (look at the implementation, and you'll get how simple it is and doesn't require any special json parser). With newline, the parser must be more complex because you need to parse progressively and keep track of the "parsing state", unless you assume that the user will never send a pretty printed json...

* On the other side there is the [SSE spec by W3C](https://www.w3.org/TR/eventsource/) sends events as newline delimited, but they assume that the payload never contains a new line (keep in mind that the sse spec is not specific to json, you could send a json but it must be on a single line)